lib/elixir/lib/task/supervised.ex

```diff
@@ -215,7 +215,7 @@ defmodule Task.Supervised do
           case waiting do
             %{^position => {_, {:ok, _} = ok}} -> Map.put(waiting, position, {nil, ok})
             %{^position => {_, :running}} -> Map.put(waiting, position, {nil, {:exit, reason}})
-            %{^position => {_, :timed_out}} -> Map.put(waiting, position, {nil, {:exit, :killed}})
+            %{^position => {_, :timed_out}} -> Map.put(waiting, position, {nil, {:exit, :timed_out}})
```

Comment:
`GenServer.start_link` does `{:error, :timeout}` : https://github.com/erlang/otp/blob/cd412d911efbda23e7dd3aef5cf910defc886211/lib/stdlib/src/proc_lib.erl#L350

Should we use `:timeout` as well?